app/src/androidTest/java/com/ea/themovie/MainActivityAndroidTest.java

```diff
@@ -0,0 +1,116 @@
+package com.ea.themovie;
+
+import android.content.Intent;
+import android.support.test.espresso.IdlingRegistry;
+import android.support.test.espresso.action.ViewActions;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+import android.support.v4.view.ViewPager;
+import android.text.TextUtils;
+
+
+import com.ea.themovie.util.RecyclerViewItemCountAssertion;
+import com.ea.themovie.util.TestData;
+import com.ea.themovie.util.ViewPagerIdlingResource;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.hasDescendant;
+import static android.support.test.espresso.matcher.ViewMatchers.isCompletelyDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static android.support.test.internal.util.Checks.checkNotNull;
+import static com.ea.themovie.util.TestUtil.atPosition;
+import static org.hamcrest.Matchers.hasEntry;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.core.AllOf.allOf;
+
+@RunWith(AndroidJUnit4.class)
+public class MainActivityAndroidTest {
+
+    Intent intent;
+
+    @Rule
+    public TestName testName = new TestName();
+
+    @Rule
+//    public ActivityTestRule<MainActivity> mMainActivityRule = new ActivityTestRule<>(MainActivity.class, false, false);
+    public ActivityTestRule<MainActivity> mMainActivityRule = new ActivityTestRule<MainActivity>(MainActivity.class){
+        @Override
+        protected void beforeActivityLaunched() {
+            super.beforeActivityLaunched();
+            setupDataTest();
+        }
+    };
+
+    @Before
+    public void setup() {
```

Comment:
If we don't have any setup, just remove this function.

Reply:
Yes, i will